src/SpoutReader.php

```diff
@@ -0,0 +1,245 @@
+        } else {
+            // Else just return the column values
+            return $row;
+        }
+    }
+
+    /**
+     * Get column headers
+     *
+     * @return array
+     */
+    public function getColumnHeaders()
+    {
+        return $this->columnHeaders;
+    }
+
+    /**
+     * Set column headers
+     *
+     * @param array $columnHeaders
+     */
+    public function setColumnHeaders(array $columnHeaders)
+    {
+        $this->columnHeaders = $columnHeaders;
+    }
+
+    /**
+     * Rewind the file pointer
+     *
+     * If a header row has been set, the pointer is set just below the header
+     * row. That way, when you iterate over the rows, that header row is
+     * skipped.
+     */
+    public function rewind()
+    {
+        $this->sheet->getRowIterator()->rewind();
+        if (null !== $this->headerRowNumber) {
+            $this->seekIndex($this->headerRowNumber + 2);
+        }
+    }
+
+    /**
+     * Set header row number
+     *
+     * @param integer $rowNumber Number of the row that contains column header names
+     */
+    public function setHeaderRowNumber($rowNumber)
+    {
+        $rowNumber = (int) $rowNumber;
+        $this->seekIndex($rowNumber + 1);
+        $this->columnHeaders = $this->sheet->getRowIterator()->current();
+        $this->headerRowNumber = $rowNumber;
+        $this->sheet->getRowIterator()->next();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function next()
+    {
+        $this->sheet->getRowIterator()->next();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function valid()
+    {
+        return $this->sheet->getRowIterator()->valid();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function key()
+    {
+        return $this->sheet->getRowIterator()->key();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function seek($position)
+    {
+        $positionIndex = $position + 1;
+        if (null !== $this->headerRowNumber) {
+            $positionIndex += $this->headerRowNumber + 1;
+        }
+
+        $this->seekIndex($positionIndex);
+    }
+
+    /**
+     * Seeks to a row index (one-based)
+     *
+     * @param $index
+     */
+    private function seekIndex($index)
+    {
+        $rowIterator = $this->sheet->getRowIterator();
+        foreach ($rowIterator as $rowIndex => $row) {
+            if ($rowIndex === $index) {
+                return;
+            }
+        }
+
+        throw new OutOfBoundsException(sprintf('Row number %d is out of range', $index));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        $count = iterator_count($this->sheet->getRowIterator());
+
+        if (null === $this->headerRowNumber) {
+            return $count;
+        }
+
+        return $count - ($this->headerRowNumber + 1);
+    }
+
+    /**
+     * Get a row
+     *
+     * @param integer $number
+     *
+     * @return array
+     */
+    public function getRow($number)
```

Comment:
What is the use case for this method?

Reply:
Removed

I have added it because the ExcelReader implement it https://github.com/portphp/excel/blob/458d5cf105206f6391e4350ee095110d18eda090/src/ExcelReader.php#L201